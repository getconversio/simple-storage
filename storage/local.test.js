'use strict';

const fs = require('fs'),
  Local = require('./local');

const local = new Local({ localDirectory: __dirname + '/../test' });

describe('storage/local', () => {
  beforeAll(() => fs.writeFileSync(__dirname + '/../test/foo', Buffer.from('foo')));
  afterAll(() => fs.unlinkSync(__dirname + '/../test/foo'));

  describe('exists', () => {
    it('should return true if a file exists', async () => {
      expect(await local.exists('foo')).toEqual(true);
    });

    it('should return false if a file does not exists', async () => {
      expect(await local.exists('bar')).toEqual(false);
    });
  });

  describe('fetchText', () => {
    it('should return the file content', async () => {
      expect(await local.fetchText('foo')).toMatchSnapshot();
    });

    it('should throw an Error if file does not exists', async () => {
      expect.assertions(1);

      try {
        await local.fetchText('bar');
      } catch (e) {
        expect(e.message).toMatch(/no such file or directory/);
      }
    });
  });
});
