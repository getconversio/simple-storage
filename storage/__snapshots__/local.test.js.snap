// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`storage/local fetchText should return the file content 1`] = `"foo"`;
